scripts/machine-launch-script.sh

```diff
@@ -1,83 +1,260 @@
+	exit 1
+    fi
+
+    if [[ "$(uname -io)" != "x86_64 GNU/Linux" ]]; then
+	echo "::ERROR:: $0 only supports 'x86_64 GNU/Linux' not '$(uname -io)'"
+	exit 1
+    fi
+
+    OS_FLAVOR=$(grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+    OS_VERSION=$(grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
+
+    echo "machine launch script started" > machine-launchstatus
+    chmod ugo+r machine-launchstatus
+
+    # platform-specific setup 
+    case "$OS_FLAVOR" in
+	ubuntu)
+	    # default /bin/sh to bash (not dash)
+	    sudo dpkg-reconfigure -p critical dash
+	    ;;
+	centos)
+	    ;;
+	*)
+	    echo "::WARNING:: Unknown OS flavor '$OS_FLAVOR'. Unable to do platform-specific setup."
+	    ;;
+    esac
+
+
+    # everything else is platform-agnostic and could easily be expanded to Windows and/or OSX
+
+    SUDO=""
+    prefix_parent=$(dirname "$CONDA_INSTALL_PREFIX")
+    if [[ ! -e "$prefix_parent" ]]; then
+	mkdir -p "$prefix_parent" || SUDO=sudo
+    elif [[ ! -w "$prefix_parent" ]]; then
+	SUDO=sudo
+    fi
+
+    if [[ -n "$SUDO" ]]; then
+	echo "::INFO:: using 'sudo' to install conda"
+	# ensure files are read-execute for everyone
+	umask 022
+    fi
+
+    # to enable use of sudo and avoid modifying 'secure_path' in /etc/sudoers, we specify the full path to conda
+    CONDA_EXE="${CONDA_INSTALL_PREFIX}/bin/conda"
+
+    if [[ -x "$CONDA_EXE" ]]; then
+	echo "::INFO:: '$CONDA_EXE' already exists, skipping conda install"
+    else
+	wget -O install_conda.sh "$CONDA_INSTALLER"  || curl -fsSLo install_conda.sh "$CONDA_INSTALLER"
+	echo "::INFO:: installing conda to '$CONDA_INSTALL_PREFIX'"
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+    fi
+
+    # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+    # for more information on strict channel_priority
+    $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+    # By default, don't mess with people's PS1, I personally find it annoying
+    $SUDO "$CONDA_EXE" config --system --set changeps1 false
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
```

Comment:
While we are at it, should we bump to a newer python (3.10.2)?

Reply:
I know conda-forge is working through getting everything supported in 3.10.  https://conda-forge.org/status/#python310 looks like ~75% of the packages in conda-forge area available for python 3.10.  I can check.

I'd be highly in favor of also bumping the python version but didn't want to lump that in with this PR.

Reply:
Ok no problem. Future work.